examples/transactions_example/processor.go

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+// This is the processor, consuming the input topic of ingress car messages,
+// deciding what traffic light to turn green for each handled intersection,
+// and emitting the traffic light states to the output topic,
+// all using the transactional API.
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"sync"
+	"time"
+)
+
+// The processor's consumer group id.
+var processorGroupID = "go-transactions-example-processor"
+
+// intersectionState
+type intersectionState struct {
+	name        string            // Name of intersection
+	partition   int32             // Input partition
+	lightState  map[string]string // Current light states, indexed by road
+	carsWaiting map[string]int    // Numbers of cars waiting, indexed by road
+	carCnt      int               // Total number of cars waiting
+	currGreen   string            // Currently green road
+	lastChange  time.Time         // Time of last light change
+}
+
+// intersectionStates maintains state per intersection
+var intersectionStates map[string]*intersectionState
```

Comment:
this state is not persisted, but that seems unwise because (if i understood the code logic correctly) in the worst case it could result in some v. unhappy cars waiting at intersections forever.

Reply:
Persisted across rebalances you mean? Correct, I wanted to provide a simple state-less processor example.

In the case no new input messages (cars) are coming in, the state machine will arbiter over the traffic lights anyway and traffic will pass, just not optimized for queue depth.